### PR TITLE
Fr us 13

### LINKED ISSUE #16: User Story 13, User can Login
As a visitor
When I visit the login path
I see a field to enter my email address and password
When I submit valid information
If I am a regular user, I am redirected to my profile page
If I am a merchant user, I am redirected to my merchant dashboard page
If I am an admin user, I am redirected to my admin dashboard page
And I see a flash message that I am logged in

### PR DESCRIPTION
**Pull request by fredrondina**
Completed US 13 + Tests. changed default user role from 0 to 1, so that logged in users are no longer considered visitors. Added name-spaced dashboard controllers and views for merchant and admin. Added dynamic_redirect method in session controller, redirects based in authenticated users role.

Closes #16

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -8,11 +8,16 @@ def create
...
       render :new
     end
   end
 
-end
\ No newline at end of file
+  def dynamic_redirect
+    redirect_to "/profile" if current_user.role == "default"
+    redirect_to "/merchant/dashboard" if current_user.role == "merchant"
+    redirect_to "/admin/dashboard" if current_user.role == "admin"
```
Comment (brian-greeson) [inline app/controllers/sessions_controller.rb:22]: I like this. How about if we make the `dynamic_redirect` method `private`. Probably not a good idea to expose those redirects outside the controller.

File: app/views/admin/dashboard/index.html.erb [hunk 2]
```diff
@@ -0,0 +1 @@
+<h1>This is a Placeholder for Admin Dashboard</h1>
```
Comment (brian-greeson) [inline app/views/admin/dashboard/index.html.erb:1]: This is a comment for your placeholder

File: db/migrate/20200407212050_add_column_role_to_users.rb [hunk 3]
```diff
@@ -1,5 +1,5 @@
 class AddColumnRoleToUsers < ActiveRecord::Migration[5.1]
   def change
-    add_column :users, :role, :integer, default: 0
+    add_column :users, :role, :integer, default: 1
```
Comment (brian-greeson) [inline db/migrate/20200407212050_add_column_role_to_users.rb:3]: Why did you change the default role?

**Review by brian-greeson — CHANGES_REQUESTED** [review]
Looks pretty good. Please resolve my comments and we'll get this merged.

File: db/migrate/20200407212050_add_column_role_to_users.rb (hunk 3 above)
Comment (fredrondina) [reply db/migrate/20200407212050_add_column_role_to_users.rb:3]: the fact that a user object exists means they are at least role :1 (default). I think it makes sense to make the default user_role 1,

File: app/controllers/sessions_controller.rb (hunk 1 above)
Comment (fredrondina) [reply app/controllers/sessions_controller.rb:22]: good idea, will make the private

**Review by brian-greeson — APPROVED** [review]
Changes look good! Merging this now

File: app/controllers/sessions_controller.rb [hunk 4]
```diff
@@ -8,11 +8,17 @@ def create
     if user.authenticate(params[:password])
       session[:user_id] = user.id
       flash[:success] = "Welcome #{user.user_detail.name}"
-      redirect_to '/'
+      dynamic_redirect
```
Comment (kelshadarby) [inline app/controllers/sessions_controller.rb:11]: Nice, this is awesome
